tests/main/snapd-snap-transition-auto-install/task.yaml

```diff
@@ -0,0 +1,126 @@
+summary: Ensure auto-install based transition to snapd snap works.
+
+details: |
+    Ensure that we can transition classic systems to snapd snap
+    with auto-install of snapd snap from snapd deb or snapd in core
+    triggered by any snap install. This change was introduced in PR14173.
+
+# Exclude Ubuntu Core systems - require proper remodeling
+# Simplify test by limiting system to ubuntu 22.04 amd64
+systems: [ubuntu-22.04-64]
+
+environment:
+    SNAPD_SRC/deb:  "deb"
+    SNAPD_SRC/core: "core"
+
+prepare: |
+    # remove all snaps to remove snapd
+    snap remove core
+    # this is only possible when snapd is the only installed snap
+    snap remove snapd
+    # at this point we expect to be running from the deb 
+    systemctl restart snapd.service
+
+    # Install deb + core before introduction of snapd auto-install
+    # on classic systems.
+    # deb:  snapd | Aug 2024 | 2.63+22.04ubuntu0.1            | http://archive.ubuntu.com/ubuntu/pool/main/s/snapd/2.63+22.04ubuntu0.1_amd64.deb
+    # deb:  snapd | Apr 2022 | 2.55.3+22.04                   | http://archive.ubuntu.com/ubuntu/pool/main/s/snapd/2.55.3+22.04_amd64.deb
+    # snap: core  | Jun 2024 | 16-2.61.4-20240607 (rev 17200) | https://snapcraft.io/core/releases
+    
+    # 'deb' : transition snapd deb to re-exec from snapd snap
+    # 'core': transition re-exec from snapd in core snap to re-exec from snapd snap
+    if [ "$SNAPD_SRC" = "deb" ]; then
+        # core < deb, exec snapd deb
+        apt install -y --allow-downgrades snapd=2.63+22.04ubuntu0.1
+    else
+        # core => deb, re-exec snapd from core snap
+        apt install -y --allow-downgrades snapd=2.55.3+22.04
+    fi
+    snap install core --revision=17200 # 16-2.61.4-20240607
+
+execute: | 
+    # check re-exec disabled, exec from snapd deb or re-exec from core snap
+    snap_mount_dir="$(os.paths snap-mount-dir)"
+    expect=""
+    if [ "${SNAP_REEXEC:-}" = "0" ]; then
+        echo "Ensure re-exec disabled"
+        expect="DEBUG: re-exec disabled by user"
+    elif [ "$SNAPD_SRC" = "deb" ]; then
+        echo "Ensure exec from snapd deb"
+        expect="DEBUG: snap \(at \"$snap_mount_dir/core/current\"\) is older \(\"2\.61\.4\"\) than distribution package \(\"2\.63\+22\.04ubuntu0\.1\"\)"
+    else
+        echo "Ensure re-exec from core snap"
+        expect="DEBUG: restarting into \"$snap_mount_dir/core/current/usr/bin/snap\""
+    fi
+    SNAPD_DEBUG=1 snap list 2>&1 | MATCH "$expect"
+
+    # Install snaps with snaps with complex interface requirements.
+    # Use specific revisions to pin expected behaviour.
+    install_snaps() {
+        snap install firefox --revision=4848      # 130.0-2
```

Comment:
maybe drop firefox? it's large, has many interfaces and for sure will pull in additional content snap dependencies

Reply:
Can do, if its ok, will wait a bit to gather more input on what constitutes a solid check for interface "migration".